### PR TITLE
Add cilium network policies

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/23014
Towards https://github.com/giantswarm/roadmap/issues/1949

### CODE REVIEW

File: hack/custom-patches.sh
Comment (kopiczko) [inline hack/custom-patches.sh:1]: It feels like this file was full of copy/paste error. E.g. it tried to call `yq` in `./bin/yq` which isn't installed by anything. I took a shortcut and set it to `yq`.

Comment (kopiczko) [reply hack/custom-patches.sh:1]: CC @jkremser

**Review by jkremser — APPROVED** [review]
lgtm

File: hack/custom-patches.sh
Comment (jkremser) [reply hack/custom-patches.sh:1]: thanks for fixing this, it went it under the radar ;)

File: helm/cluster-api-provider-cloud-director/templates/networking.k8s.io_v1_networkpolicy_capvcd-controller-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      protocol: TCP
+    - port: 1053
+      protocol: TCP
+    to:
+    - namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: kube-system
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/name: coredns
```
Comment (jkremser) [inline helm/cluster-api-provider-cloud-director/templates/networking.k8s.io_v1_networkpolicy_capvcd-controller-manager.yaml:33]: kubelet and image pulling from various registries should not be controlled by cillium network policies, right? If we ever gonna use the virtual kubelet (running as pod), then we should start care.

Comment (kopiczko) [reply helm/cluster-api-provider-cloud-director/templates/networking.k8s.io_v1_networkpolicy_capvcd-controller-manager.yaml:33]: Image pulling is done by containerd I think which is runnign outside kubernetes. Same goes for kubelet - it's running outside kubernetes.

**Pull request merged**